src/main/groovy/com/ullink/NuGetPack.groovy

```diff
@@ -122,7 +123,8 @@ class NuGetPack extends BaseNuGet {
 
     File getPackageFile() {
         def spec = getNuspec()
-        def version = spec.metadata.version ?: project.version
-        new File(getDestinationDir(), spec.metadata.id.toString() + '.' + version + '.nupkg')
+        def version = spec?.metadata?.version ?: project.version
+        def id = spec?.metadata?.id?.toString() ?: FilenameUtils.removeExtension(project.msbuild.projects.find { csprojPath.equalsIgnoreCase(it.value.projectFile) }.value.dotnetAssemblyFile.name)
```

Comment:
this plugin only depends optionally on msbuild, also you shouldn't assume the task named msbuild is related to this pack operation (see in the ctor how it tries to find/guess the related msbuild task, if any)